src/panels/logbook/ha-logbook-renderer.ts

```diff
@@ -31,9 +34,30 @@ import {
 import { HomeAssistant } from "../../types";
 import { brandsUrl } from "../../util/brands-url";
 
-const EVENT_LOCALIZE_MAP = {
-  script_started: "from_script",
+const triggerDomains = ["script", "automation"];
+const triggerPhrases = {
+  "triggered by ": "triggered_by",
+  "state of ": "state_of",
+  "send command ": "send_command",
+};
+
+const localizeTriggerMessages = (hass: HomeAssistant, message: string) => {
+  for (const triggerPhrase in triggerPhrases) {
+    if (message.startsWith(triggerPhrase)) {
+      message = message.replace(
+        triggerPhrase,
+        `${hass.localize(
+          `ui.components.logbook.${triggerPhrases[triggerPhrase]}`
+        )} `
+      );
```

Comment:
The problem with localization is that we can't assume that words are all the same order. So it would need to be that we extract the "other part" and inject that as a  variable. 

Something like this (untested):

```suggestion
      message = hass.localize(
        `ui.components.logbook.${triggerPhrases[triggerPhrase]}`,
        {what: message.substr(triggerPhrase)}
      )
```

Reply:
Now each described event can be triggered by different things. So it will be wild to try to create a localized string for every permutation, which would be necessary if we want to do it 100% correct.

We could also split it up in the UI such that we don't have to create localized strings for each permutation of trigger combined with each cause.

Reply:
btw I still think that this PR is an improvement and we should go for it

Reply:
I picked the 3 that appear the most common since it will get most of the cases.

I don't think we can ever get to 100% without heroic effort and singular dedication to this.  It would take an army to get there!

Reply:
These are the ones its trying to pick up

https://github.com/home-assistant/core/blob/dev/homeassistant/components/homeassistant/triggers/state.py#L182
https://github.com/home-assistant/core/blob/dev/homeassistant/components/homeassistant/triggers/numeric_state.py#L159
https://github.com/home-assistant/core/blob/dev/homeassistant/components/homeassistant/triggers/time.py#L57
https://github.com/home-assistant/core/blob/dev/homeassistant/components/homeassistant/triggers/time_pattern.py#L82
https://github.com/home-assistant/core/blob/dev/homeassistant/components/homeassistant/triggers/homeassistant.py#L39
https://github.com/home-assistant/core/blob/dev/homeassistant/components/homeassistant/triggers/homeassistant.py#L57
https://github.com/home-assistant/core/blob/dev/homeassistant/components/homeassistant/triggers/event.py#L109

Maybe we make we also make these keys?

Reply:
I think we already have enough to do that since context_source is that without the 'triggered by' https://github.com/home-assistant/core/blob/dev/homeassistant/components/automation/logbook.py

So we can make all the source strings to a locale key and use that when available

I think that was actually the original plan but we never got around to implementing on the frontend

Reply:
https://github.com/home-assistant/core/blob/9c3f9491651f409e8b4d0d645115b55b14f06165/homeassistant/components/homeassistant/triggers/time.py?rgh-link-date=2022-05-22T05%3A58%3A48Z#L163
https://github.com/home-assistant/core/blob/9c3f9491651f409e8b4d0d645115b55b14f06165/homeassistant/components/homeassistant/triggers/time.py?rgh-link-date=2022-05-22T05%3A58%3A48Z#L121

Reply:
Last push contained the scope to only the trigger messages since thats 99% of what I see in the logbook anyways